### DISCUSSION TITLE
PIWO based dblp tool

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [SmartDataAnalytics/OpenResearch#112] [Ideas]
After Discussion with Marcel Ackermann  a few weeks ago that there is bibtex compatible bibliographic reference available on dblp i created a small tool based on the [MediaWiki Extension Piwo](https://www.mediawiki.org/wiki/Extension:Piwo). The idea is that you can create a SemanticMediaWiki reference with a direct link to the event.

Here is an example how it works.

1. Start with a paper reference such as https://www.researchgate.net/publication/329539435_The_Scientific_Events_Ontology_of_The_OpenResearchorg_Curation_Platform
2. Lookup the DOI https://dl.acm.org/doi/10.1145/3297280.3297631
3. enter the doi in the dblp search
![grafik](https://user-images.githubusercontent.com/1336221/108360742-de907c00-71f1-11eb-9215-34de43c4eefc.png)
4. find the dblp pseudo-PID in it's bibtex record https://dblp.org/rec/conf/sac/FathallaVA019.html?view=bibtex
5. feed it into the PIWO based MediaWiki template ```{{dblp|/conf/sac/FathallaVA019}}```
6. cut and paste the result in to the SMW based MediaWiki page where you'd like to reference the paper e.g.  https://rq.bitplan.com/index.php/SEO 
```[[CiteRef::DBLP:conf/sac/FathallaVA019]]

{{#scite:DBLP:conf/sac/FathallaVA019
|type=article
|authors="Said Fathalla and Sahar Vahdati and Sören Auer and Christoph Lange"
|editor=Chih{-Cheng Hung and George A. Papadopoulos}
|title=The scientific events ontology of the OpenResearch.org curation platform
|booktitle=Proceedings of the 34th {ACM/SIGAPP Symposium on Applied Computing
|{SAC} 2019, Limassol, Cyprus, April 8-12, 2019}
|pages=2311--2313
|publisher={ACM}
|year=2019
|url=https://doi.org/10.1145/3297280.3297631
|doi=10.1145/3297280.3297631
|timestamp=Mon, 15 Jun 2020 17:05:37 +0200
|biburl=https://dblp.org/rec/conf/sac/FathallaVA019.bib
|bibsource=dblp computer science bibliography, https://dblp.org
}} 
```

add the acronym of the conference
```
|acronym=SAC 2019
```
7. The reference shows and the event "SAC 2019" is clickable
![grafik](https://user-images.githubusercontent.com/1336221/108361262-86a64500-71f2-11eb-91b0-b6c3dcc191c0.png)
8. you end up in the prototype conference lookup wiki https://cr.bitplan.com/index.php/SAC_2019 
![grafik](https://user-images.githubusercontent.com/1336221/108361418-c0774b80-71f2-11eb-8b14-b3f60e45acd8.png)
9. and you can continue at [Proceedings Title Parser result for SAC 2019](http://ptp.bitplan.com/parse?titles=SAC+2019%0D%0A&examples=example1&format=html&metadebug=on)

I'd love to make this all a one step process and e.g. integrate with https://confident.dbis.rwth-aachen.de/dblpconf/series/sac

If you'd like to get the PIWO python source code for this just contact me.

**Comment by mra42** [comment]
A note on this: If the DOI is known, you should be able to get the data more directly via our DOI API, for example via Content Negotiation and Access-Header:

https://dblp.org/doi/10.1145/3297280.3297631

or  directly via file extension:

https://dblp.org/doi/10.1145/3297280.3297631.html
https://dblp.org/doi/10.1145/3297280.3297631.bib
https://dblp.org/doi/10.1145/3297280.3297631.ris
https://dblp.org/doi/10.1145/3297280.3297631.nt
https://dblp.org/doi/10.1145/3297280.3297631.rdf
https://dblp.org/doi/10.1145/3297280.3297631.xml

The DOI part is case-insensitive. If the DOI is unknown to dblp, a status 404 is responded.

**Comment by WolfgangFahl** [comment]
Thx for the hint - i appreciate it!
